src/main/java/org/folio/service/pieces/flows/update/PieceUpdateFlowInventoryManager.java

```diff
@@ -93,72 +90,70 @@ private Future<Location> handleHolding(PieceUpdateHolder holder, RequestContext
-                            pieceToUpdate.setHoldingId(holdingId);
-                            location.setLocationId(null);
-                            location.setHoldingId(holdingId);
-                          });
-                          return location;
-                        });
+        .map(holdingId -> {
+          Optional.ofNullable(holdingId).ifPresent(holdingIdP -> {
+            pieceToUpdate.setLocationId(null);
+            pieceToUpdate.setHoldingId(holdingId);
+            location.setLocationId(null);
+            location.setHoldingId(holdingId);
+          });
+          return location;
+        });
     }
     return Future.succeededFuture(new Location().withLocationId(pieceToUpdate.getLocationId()));
   }
 
   private Future<String> handleItem(PieceUpdateHolder holder, RequestContext requestContext) {
     CompositePoLine poLineToSave = holder.getPoLineToSave();
     Piece pieceToUpdate = holder.getPieceToUpdate();
-    if (DefaultPieceFlowsValidator.isCreateItemForPiecePossible(pieceToUpdate, poLineToSave)) {
-      return inventoryManager.getItemRecordById(pieceToUpdate.getItemId(), true, requestContext)
-        .compose(jsonItem -> {
-        if (holder.isCreateItem() && (jsonItem == null || jsonItem.isEmpty()) && pieceToUpdate.getHoldingId() != null) {
+    if (!DefaultPieceFlowsValidator.isCreateItemForPiecePossible(pieceToUpdate, poLineToSave)) {
+        return Future.succeededFuture();
+    }
+    return inventoryManager.getItemRecordById(pieceToUpdate.getItemId(), true, requestContext)
+      .compose(jsonItem -> {
+        boolean noJsonItem = jsonItem == null || jsonItem.isEmpty();
+        if (holder.isCreateItem() && noJsonItem && pieceToUpdate.getHoldingId() != null) {
           return pieceUpdateInventoryService.manualPieceFlowCreateItemRecord(pieceToUpdate, poLineToSave, requestContext);
-        } else if (jsonItem != null && !jsonItem.isEmpty()) {
-          return updateItemWithFields(jsonItem, poLineToSave, pieceToUpdate).compose(
-            aVoid -> inventoryManager.updateItem(jsonItem, requestContext).map(item -> jsonItem.getString(ID)));
-        } else {
-          return Future.succeededFuture();
         }
+        if (!noJsonItem) {
+          return updateItemWithFields(jsonItem, poLineToSave, pieceToUpdate)
+            .compose(ignored -> inventoryManager.updateItem(jsonItem, requestContext).map(item -> jsonItem.getString(ID)));
+        }
+        return Future.succeededFuture();
       });
-    }
-    return Future.succeededFuture();
   }
 
   private Future<Void> updateItemWithFields(JsonObject item, CompositePoLine compPOL, Piece piece) {
     Optional.ofNullable(piece.getHoldingId())
       .ifPresent(pieceHoldingId -> item.put(ITEM_HOLDINGS_RECORD_ID, piece.getHoldingId()));
-    Optional.ofNullable(piece.getCopyNumber())
-      .ifPresentOrElse(copyNumber -> item.put(COPY_NUMBER, copyNumber), () -> item.remove(COPY_NUMBER));
```

Comment:
can you provide details why this line is deleted?

Reply:
This is another place in the same method call chain where we overwrite item fields:
`PieceUpdateFlowInventoryManager::processInventory` -> `packagePoLineUpdateInventory` -> `handleItem` -> `updateItemWithFields`

Reply:
And `holdingId` itselft is defined in `handleHolding` method so it will not make sense to replace item `hloldinId` update before calling this method.